src/main/java/com/marklogic/appdeployer/command/ResourceManager.java

```diff
@@ -0,0 +1,18 @@
+package com.marklogic.appdeployer.command;
+
+import java.io.File;
+
+/**
+ * Defines operations for managing whether a module needs to be installed or not.
+ */
+public interface ResourceManager {
+
+    /**
+     * Give the implementor a chance to initialize itself - e.g. loading data from a properties file or other resource.
+     */
+    void initialize();
+
+    boolean hasFileBeenModifiedSinceLastDeployed(File file);
+
+    void saveLastDeployedHash(File file);
```

Comment:
I think this should have a name of "afterResourceFileDeployed". The client doesn't care about hashes.